rfcs/unsealed-table-literals.md

```diff
@@ -0,0 +1,74 @@
+# Unsealed table literals
+
+## Summary
+
+Currently the only way to create an unsealed table is as an empty table literal `{}`.
+This RFC proposes making all table literals unsealed.
+
+## Motivation
+
+Table types can be *sealed* or *unsealed*. These are different in that:
+
+* Unsealed table types are *precise*: if a table has unsealed type `{ p: number, q: string }`
+  then it is guaranteed to have only properties `p` and `q`.
+
+* Sealed tables support *width subtyping*: if a table has sealed type `{ p: number }`
+  then it is guaranteed to have at least property `p`, so we allow `{ p: number, q: string }`
+  to be treated as a subtype of `{ p: number }`
+
+* Unsealed tables can have properties added to them: if `t` has unsealed type
+  `{ p: number }` then after the assignment `t.q = "hi"`, `t`'s type is updated to be
+  `{ p: number, q: string }`.  
+
+* Unsealed tables are subtypes of sealed tables.
+
+Currently the only way to create an unsealed table is using an empty table literal, so
+```lua
+  local t = {}
+  t.p = 5
+  t.q = "hi"
+```
+typechecks, but
+```lua
+  local t = { p = 5 }
+  t.q = "hi"
+```
+does not.
+
+This causes problems in examples, for example
```

Comment:
I think this needs some rewording

Reply:
For the repeated "example" you mean? If so, fixed.